src/tests/test_targets.py

```diff
@@ -43,3 +44,15 @@ def test_delete_clusters(self, api_client: InventoryClient, cluster_configuratio
             cluster.delete()
 
         log.info(f"Successfully deleted {len(clusters)} clusters")
+
+    @JunitTestSuite()
+    def test_destroy_terraform(
+        self,
+        prepared_controller_configuration: BaseNodesConfig,
+        controller: NodeController,
+    ):
+        """Destroy cluster via terraform"""
+        kwargs = prepared_controller_configuration.get_all()
+        controller._tf.set_vars(**kwargs)
```

Comment:
`_tf` is not NodeController Attribute 

```AttributeError: 'TerraformController' object has no attribute '_tf'```

Reply:
`NutanixController`/`VSphereController` are different than the `TerraformController`.
There is a task for create some kind of interface for all of the terraform controllers but it's not prioritized

Reply:
Also after removing the underscore it seems that it won't destroy the nodes

Reply:
We need to somehow restore the controllers state and destroy all of the nodes

Reply:
Created `get_terraform_controller` which returns `TerraformController`. Later on the previous fixture can be replaced with it too, but I'd rather copy code instead of breaking things